### PR TITLE
[depends] bump libnfs to 4.0.0

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Bumps libnfs to 4.0.0 

## Motivation and Context
We are running an old version 2.0.0. 4.0.0 brings a slew of fixes/updates including nfs4 support in the lib. This will require some core changes to potentially enable this however.

## How Has This Been Tested?
Build test for all depends platforms. Runtime tested macos.
Might be worth trying to get some community testing before merge.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: tools/depends/target/libnfs/Makefile [hunk 1]
```diff
@@ -3,12 +3,14 @@ DEPS= ../../Makefile.include Makefile
 
 # lib name, version
 LIBNAME=libnfs
-VERSION=2.0.0
+VERSION=4.0.0
 SOURCE=$(LIBNAME)-$(VERSION)
 ARCHIVE=$(SOURCE).tar.gz
 
+BASE_URL := https://github.com/sahlberg/libnfs/archive
```
Comment (Rechi) [inline tools/depends/target/libnfs/Makefile]: This is wrong.

**Review by Rechi — CHANGES_REQUESTED** [review]

File: tools/depends/target/libnfs/Makefile (hunk 1 above)
Comment (fuzzard) [reply tools/depends/target/libnfs/Makefile]: Yeah, will update that, need to get the source tar on mirrors and drop this

Comment (AlwinEsch) [reply tools/depends/target/libnfs/Makefile]: Hace it uploaded to our ftp and available now on http://mirrors.kodi.tv/build-deps/sources/libnfs-4.0.0.tar.gz

**Comment by howie-f** [comment]
fyi, testing this pr but had no luck connecting to nfsv4 from fire-tv. am i right in the assumption this should work then, or am i missing something?

**Comment by MilhouseVH** [comment]
From [libnfs](https://github.com/sahlberg/libnfs)
```
NFSv4:
======
NFSv3 is the default but NFSv4 can be selected either by using the URL argument
version=4 or programatically calling nfs_set_version(nfs, NFS_V4) before
connecting to the server/share.
```
so it's unlikely Kodi supports NFSv4 without additional changes - maybe it's possible to pass `version=4` on the `nfs://` url?

**Comment by MilhouseVH** [comment]
Try `nfs://server/path?version=4`

**Comment by howie-f** [comment]
From libnfs: `nfs://<server|ipv4|ipv6>/path[?arg=val[&arg=val]*]`
>Try nfs://server/path?version=4

tried this with and without asterisk as well... no luck.
thx @MilhouseVH, not intended to start a support thread here ;)

File: tools/depends/target/libnfs/Makefile (hunk 1 above)
Comment (fuzzard) [reply tools/depends/target/libnfs/Makefile]: Thanks Alwin, I'll fix this up today

**Review by wsnipex — APPROVED** [review]

**Comment by AkariDN** [comment]
About Kodi current NFS implementation:
* It doesn't take into account URL options, so some code changes are required to pass ```version=4``` from url to libnfs.
* It uses deprecated ```nfs_stat``` function. We should use ```nfs_stat64``` with new library (in my case old function doesn't work at all).
* It always retrieves exports from ```mountd``` (like ```showmount -e```), so we can't use NFS4 without ```rpcbind``` and public ```mountd```. (```rpcbind``` is not required for NFS4 and ```mountd``` can be started locally - only one port is used). There should be an option to avoid using ```mountd```.

Moreover, it seems that Kodi is not well prepared for URLs with options. I've found at least two design bugs:
* Video database stores pathes without options (see ```CVideoDatabase::SplitPath```) but loads with options (see ```CVideoDatabase::GetPathId```). Watched state and other stuff will not work for such URLs without appropriate code changes.
* Options in URL can be unexpectedly reordered, but some comparison functions don't take it into accout. See ```CGUIDialogContextMenu::GetShare``` for example. Context menu on shares with options may not work properly. (Share created with ```version=4&norpc``` options may unexpectedly become ```norpc&version=4```)

Here are fast code changes to make NFS4 work (tested only on my system) with bugfixes:
https://github.com/AkariDN/xbmc/commit/24f6a8875823229bec22d0551bb2771c46d5b32d
URLs like ```nfs://myserver/myexport/?version=4&norpc``` can be used for shares, where all options are passed to libnfs (```version=4``` for NFS4) and ```norpc``` is a special option to avoid calls to ```mountd```.
You are free to use this code as you wish.

**Comment by howie-f** [comment]
not sure at all - maybe this interferes/should be taken care of in vfs.nfs ??

**Comment by fuzzard** [comment]
There's no intention to continue down the path of VFS.nfs/SMB at this point in time. From what I'm told, difficulties around add-on startup/loading and ordering of other things that require FS calls.

**Comment by fuzzard** [comment]
@peak3d have added a commit that bumps libnfs to a hash commited Jun 9th 2020 to support your timeout PR #15686

Builds ok on darwin_embedded, Maybe you want to try on android with your timeout PR if you find time?

**Comment by fuzzard** [comment]
Jenkins build this please

**Comment by peak3d** [comment]
@fuzzard works as expected on Android.

**Comment by fuzzard** [comment]
Jenkins build this please

**Pull request merged**